### PR TITLE
Add docs for thread fields

### PR DESCRIPTION
**Pull request by mattvagni**
As part of this I also made the typescript code examples more consistent where the client is consistently imported and shown.

### CODE REVIEW

File: _snippets/graphql/delete-thread-field.mdx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      }
+    }
+  }
+}
+```
+
+```json Variables
+{
+  "input": {
+    "threadFieldId": "th_01HVNWFJS395XVPPBJNE6A8BHP"
```
Comment (preslavmihaylov) [inline _snippets/graphql/delete-thread-field.mdx:21]: the id is of a thread here, it should be `tf_...`

File: _snippets/graphql/upsert-thread-field.mdx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+```json Variables
+{
+  "input": {
+    "identifier": {
+      "threadId": "th_01HVNWFJS395XVPPBJNE6A8BHP",
+      "key": "product_area"
+    },
+    "type": "STRING",
+    "stringValue": "security"
```
Comment (preslavmihaylov) [inline _snippets/graphql/upsert-thread-field.mdx:42]: `booleanValue` should be set to `null`

File: _snippets/typescript-sdk/delete-customer.mdx [hunk 3]
```diff
@@ -1,9 +1,7 @@
-```tsx
-import { PlainClient } from '@team-plain/typescript-sdk';
+```ts
+import { PlainClient, uiComponent } from '@team-plain/typescript-sdk';
```
Comment (preslavmihaylov) [inline _snippets/typescript-sdk/delete-customer.mdx:2]: `uiComponent` is imported but not used, this is in couple of places.

File: api-reference/graphql/threads/thread-fields.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+title: 'Thread fields'
+---
+
+Thread fields allow you to extend Plain's thread data model. The thread fields which you want to support have to confirm with a schema configured in **Settings** → **Thread fields**.
+
+Thread fields can be nested and be either a boolean, text or a string enum.
+
+Thread fields can be required. When they are required, their value must be set in order for the thread to be marked as done.
+
+For interacting with thread fields via the API, every field has a `key` defined in it's schema. Keys make it possible to quickly refer to a thread field without having to know it's ID in the schema. For example if you have a field called "Product Area" the key you might choose for the key to be `product_area`.
```
Comment (preslavmihaylov) [inline api-reference/graphql/threads/thread-fields.mdx:11]:
```suggestion
For interacting with thread fields via the API, every field has a `key` defined in its schema. Keys make it possible to quickly refer to a thread field without having to know its ID in the schema. For example if you have a field called "Product Area" the key you might choose for the key to be `product_area`.
```

File: thread-fields.mdx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+- A boolean (true or false)
+- Text
+- A dropdown (single select from a predetermined list)
+
+Thread fields can also be nested and be conditional on answers from other fields.
+
+Fields can also be marked as required. This is useful when there is some information you want to collect on every support request before it is marked as done.
+
+Thread fields can also be **auto-filled via AI**. To enable this behaviour in the settings for any thread field enable "Autofill via AI".
```
Comment (preslavmihaylov) [inline thread-fields.mdx:20]: could be worth mentioning that the thread field's description is used as context for the AI, so advise is to add that if autofill AI is enabled

**Review by presmihaylov — APPROVED** [review]

**Pull request merged**